pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,373 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2014 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "jquery",
+    "base1/cockpit",
+    "kubernetes/client",
+    "base1/mustache"
+], function($, cockpit,kubernetes,Mustache) {
+    "use strict";
+
+    var appdeployer = {};
+    var client = kubernetes.k8client();
+
+    function debug() {
+        if (window.debugging == "all" || window.debugging == "deploy")
+            console.debug.apply(console, arguments);
+    }
+
+    function failure(ex) {
+        console.warn(ex);
+    }
+
+    function isJsonString(str) {
+        try {
+            JSON.parse(str);
+        } catch (e) {
+            return false;
+        }
+        return true;
+    }
+
+    function get_kentities(entity_key) {
+        var elist = [];
+        if (client[entity_key]) {
+            var el = client[entity_key];
+            for (var i = 0; i < el.length; i++) {
+                elist.push(el[i].metadata.name);
+            }
+        }
+        return elist;
+    }
+
+    function deploy_manager() {
+
+
+        var is_deploying = $('#deploy-app-deploying');
+        var deploy_notification_success = $('#deploy-app-notification-success-template').html();
+        var deploy_notification_failure = $('#deploy-app-notification-failure-template').html();
+        Mustache.parse(deploy_notification_success);
+        Mustache.parse(deploy_notification_failure);
+
+        function deploy_app(namespace, jsonData) {
+            if (action_in_progress()) {
+                console.log('Unable to Deploy app at this time because a call to deploy manager ' +
+                    'is already in progress. Please try again.');
```

Comment:
Is this a programmer or logic error? That is, buttons should have been disabled. If so, and it's an assertion style, check, then I would suggest just:

```
throw "Action already in progress"
```